### PR TITLE
restore vector functions

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/lisp.rs [hunk 1]
```diff
@@ -106,11 +142,33 @@ impl<T> ExternalPtr<T> {
...
+    type Target = T;
+    fn deref(&self) -> &Self::Target {
+        unsafe { &*self.0 }
+    }
 }
 
-pub type LispSubrRef = ExternalPtr<Aligned_Lisp_Subr>;
-impl LispSubrRef {
-    pub fn as_mut(self) -> *mut Aligned_Lisp_Subr {
-	self.0
```
Comment (brotzeit) [inline rust_src/src/lisp.rs:114]: @DavidDeSimone I added it to `ExternalPtr<T>`

File: rust_src/src/lisp.rs [hunk 2]
```diff
@@ -106,11 +142,38 @@ impl<T> ExternalPtr<T> {
...
+impl<T> DerefMut for ExternalPtr<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { &mut *self.0 }
     }
 }
+
+// Lisp_Subr support
+
+pub type LispSubrRef = ExternalPtr<Aligned_Lisp_Subr>;
+unsafe impl Sync for Aligned_Lisp_Subr {}
```
Comment (brotzeit) [inline rust_src/src/lisp.rs:179]: This was in the old code base ?

File: rust_src/src/lisp.rs (hunk 1 above)
Comment (DavidDeSimone) [reply rust_src/src/lisp.rs:114]: That looks good

File: rust_src/src/lisp.rs (hunk 2 above)
Comment (DavidDeSimone) [reply rust_src/src/lisp.rs:179]: There was an impl Sync for Lisp_Subr in the old codebase, however that lived in remacs_sys.rs. I did add equivalent code for Aligned_Lisp_Subr implementing Sync in remacs_sys, so I don't think we need that here. We should still have `pub type LispSubrRef = ExternalPtr<Aligned_Lisp_Subr>;` here however

File: rust_src/Cargo.toml [hunk 3]
```diff
@@ -58,9 +58,7 @@ panic = "abort"
 
 [features]
 default = [
-  "use-xml2", "unexec", "window-system", "window-system-x11", "libvterm",
-
-]
+  "use-xml2", "unexec", "window-system", "window-system-x11",]
```
Comment (DavidDeSimone) [inline rust_src/Cargo.toml:61]: Do we want to remove libvterm here intentionally?

Comment (brotzeit) [reply rust_src/Cargo.toml:61]: Yeah, it's currently not in the code base.

File: rust_src/src/lisp.rs (hunk 2 above)
Comment (brotzeit) [reply rust_src/src/lisp.rs:179]: > I did add equivalent code for Aligned_Lisp_Subr implementing Sync in remacs_sys, so I don't think we need that here

Ah I missed that.

**Pull request merged**